scrapers/scraper.py

```diff
@@ -0,0 +1,434 @@
+                    "--quiet",
+                    "https://github.com/apertium/{}".format(dirName),
+                ],
+                cwd=os.path.join(path, "scrapers"),
+            )
+        shutil.copyfile(
+            os.path.join(path, "scrapers", ".mailmap"),
+            os.path.join(path, "scrapers", dirName, ".mailmap"),
+        )
+        commits = (
+            subprocess.check_output(
+                [
+                    "git",
+                    "log",
+                    "--format=%H %aN %aI",
+                    "--follow",
+                    "apertium-{0}.{0}.dix".format(dixName),
+                ],
+                cwd=os.path.join(path, "scrapers", dirName),
+            )
+            .decode("utf-8")
+            .split("\n")
+        )
+
+        try:
+            oldFile = json.load(
+                open(os.path.join(path, "json", "{}.json".format(language)), "r")
+            )
+            for data in oldFile:
+                if data["name"] == pairName:
+                    history = data["history"]
+                    break
+                history = []
+        except FileNotFoundError:
+            history = []
+
+        commits.pop()  # Last line is always empty
+        for commit in commits:
+            commitData = {}
+            data = commit.split(" ")
+            commitData["sha"] = data[0]
+            commitData["author"] = data[1]
+            commitData["date"] = data[2]
+
+            if any(commitData["sha"] == cm["sha"] for cm in history):
+                continue
+
+            dixFile = "https://raw.githubusercontent.com/apertium/apertium-{0}/{1}/apertium-{0}.{0}.dix".format(
+                pairName, commitData["sha"]
+            )
+            stems = countDixStems(dixFile, True)
+            if stems == -1:
+                dixName = iso3to2[pairList[0]] + "-" + iso3to2[pairList[1]]
+                dixFile = "https://raw.githubusercontent.com/apertium/apertium-{0}/{2}/apertium-{1}.{1}.dix".format(
+                    pairName, dixName, commitData["sha"]
+                )
+                stems = countDixStems(dixFile, True)
+                if stems == -1:
+                    logging.getLogger("pairHistory").warning(
+                        "Unable to count dix stems for {} in commit {}".format(
+                            pairName, commitData["sha"]
+                        )
+                    )
+                    continue
+
+            commitData["stems"] = stems["stems"]
+            history.append(commitData)
+
+        langPackages.append({"name": pairName, "history": history})
+    return langPackages
+
+
+def monoData(packages, updatemailmap, path):
+    """Returns data for all monolingual dictionaries: state, stems, location and contributors"""
+    data = []
+    for package in packages:
+        if not (
+            re.match(r"apertium-\w+$", package["name"])
+            and package["name"][len("apertium-") :] in languages
```

Comment:
We do this `len("apertium-") :` thing a bunch of times. Could you move it into a function?